Common/Tools/TrackTuner.h

```diff
@@ -295,330 +349,378 @@ struct TrackTuner {
-      sigma1PtSnp = trackParCov.getSigma1PtSnp();
-      if (oneOverPtCurrent > 0.)
-        sigma1PtSnp *= (oneOverPtUpgr / oneOverPtCurrent);
-      trackParCov.setCov(sigma1PtSnp, 12);
-
-      //       if(spt1o>0.)             covar[13]*=(spt1n/spt1o);//tanTpt
-      sigma1PtTgl = trackParCov.getSigma1PtTgl();
-      if (oneOverPtCurrent > 0.)
-        sigma1PtTgl *= (oneOverPtUpgr / oneOverPtCurrent);
-      trackParCov.setCov(sigma1PtTgl, 13);
+      if (dcaZResMC > 0.) {
+        sigmaTglZ *= ((dcaZResData / dcaZResMC));
+        trackParCov.setCov(sigmaTglZ, 7);
+      }
 
-      //       if(spt1o>0.)             covar[14]*=(spt1n/spt1o)*(spt1n/spt1o);//ptpt
-      sigma1Pt2 = trackParCov.getSigma1Pt2();
-      if (oneOverPtCurrent > 0.)
-        sigma1Pt2 *= (oneOverPtUpgr / oneOverPtCurrent);
-      trackParCov.setCov(sigma1Pt2, 14);
-    }
+      // checking and updating track cov matrix elements for 1/Pt begins
+      if ((updateCurvature) && (!updateCurvatureIU)) {
+        //       if(sd0rpo>0. && spt1o>0.)covar[10]*=(sd0rpn/sd0rpo)*(spt1n/spt1o);//ypt
+        sigma1PtY = trackParCov.getSigma1PtY();
+        if (dcaXYResMC > 0. && qOverPtMC > 0.) {
+          sigma1PtY *= ((dcaXYResData / dcaXYResMC) * (qOverPtData / qOverPtMC));
+          trackParCov.setCov(sigma1PtY, 10);
+        }
+
+        //       if(sd0zo>0. && spt1o>0.) covar[11]*=(sd0zn/sd0zo)*(spt1n/spt1o);//zpt
+        sigma1PtZ = trackParCov.getSigma1PtZ();
+        if (dcaZResMC > 0. && qOverPtMC > 0.) {
+          sigma1PtZ *= ((dcaZResData / dcaZResMC) * (qOverPtData / qOverPtMC));
+          trackParCov.setCov(sigma1PtZ, 11);
+        }
+
+        //       if(spt1o>0.)             covar[12]*=(spt1n/spt1o);//sinPhipt
+        sigma1PtSnp = trackParCov.getSigma1PtSnp();
+        if (qOverPtMC > 0.) {
+          sigma1PtSnp *= (qOverPtData / qOverPtMC);
+          trackParCov.setCov(sigma1PtSnp, 12);
+        }
+
+        //       if(spt1o>0.)             covar[13]*=(spt1n/spt1o);//tanTpt
+        sigma1PtTgl = trackParCov.getSigma1PtTgl();
+        if (qOverPtMC > 0.) {
+          sigma1PtTgl *= (qOverPtData / qOverPtMC);
+          trackParCov.setCov(sigma1PtTgl, 13);
+        }
+
+        //       if(spt1o>0.)             covar[14]*=(spt1n/spt1o)*(spt1n/spt1o);//ptpt
+        sigma1Pt2 = trackParCov.getSigma1Pt2();
+        if (qOverPtMC > 0.) {
+          sigma1Pt2 *= (qOverPtData / qOverPtMC);
+          trackParCov.setCov(sigma1Pt2, 14);
+        }
+      } // ---> track cov matrix elements for 1/Pt ends here
+    }   // ---> updateTrackCovMat block ends here
 
     if (updatePulls) {
-      double ratioDCAxyPulls = dcaXYPullCurrent / dcaXYPullUpgr;
-      double ratioDCAzPulls = dcaZPullCurrent / dcaZPullUpgr;
+      double ratioDCAxyPulls = 1.0;
+      double ratioDCAzPulls = 1.0;
+      if (dcaZPullData != 0.0)
+        ratioDCAzPulls = dcaZPullMC / dcaZPullData;
+      if (dcaXYPullData != 0.0)
```

Comment:
```suggestion
      if (dcaXYPullData > 0.0)
```
Never use equalities with `float`s or `double`s